src/transformers/models/mask2former/modeling_mask2former.py

```diff
@@ -1123,6 +1129,7 @@ def forward(
         attention_mask=None,
         position_embeddings=None,
         spatial_shapes=None,
+        spatial_shapes_list=None,
```

Comment:
Should we pass both the list and the tensor? Can't we convert it inside or are there any performance issues in that case?
cc @yonigozlan

Reply:
we could probably convert the list to the tensor inside this forward call without affecting performance. 

the spatial_shapes tensor is directly passed into multiple `encoder_layer`s of this module so this feels like the lowest we can go without changing the performance. Although I wonder how bad the performance impact will be if we never create a tensor and instead read the spatial_shapes_list here: https://github.com/philkuz/transformers/blob/78f384808a70d346cd8f1e8d314411510edec11d/src/transformers/models/mask2former/modeling_mask2former.py#L962

Reply:
Wrote a quick torch benchmark based on where this tensor is actually used
```
import time
import torch

# Create sample data - list of 1000 tuples with 3 elements each
# https://github.com/philkuz/transformers/blob/78f384808a70d346cd8f1e8d314411510edec11d/src/transformers/models/mask2former/modeling_mask2former.py#L1219
num_feature_levels = 3  # hardcoded in the modeling code

# https://huggingface.co/facebook/mask2former-swin-small-coco-instance/blob/main/config.json#L2118
# https://huggingface.co/facebook/mask2former-swin-large-coco-instance/blob/main/config.json#L2118
config_encoder_layers = (
    6  # Set in the config, seems to be set at 6 across small and large models
)
data = [(i, i + 1) for i in range(num_feature_levels)]

# Run benchmark
num_iterations = 1000
total_time = 0

for _ in range(num_iterations):
    start = time.perf_counter()
    tensor = torch.tensor(data)
    for i in range(config_encoder_layers):
        offset_normalizer_tensor = torch.stack([tensor[..., 1], tensor[..., 0]], -1)
    end = time.perf_counter()
    total_time += end - start

avg_time = total_time / num_iterations
print(
    f"Average time from spatial_shapes tensor {num_iterations} iterations: {avg_time:.6f} seconds"
)


num_iterations = 1000
total_time = 0
for _ in range(num_iterations):
    start = time.perf_counter()
    for i in range(config_encoder_layers):
        offset_normalizer_tensor = torch.tensor([[d[1], d[0]] for d in data])
    end = time.perf_counter()
    total_time += end - start

avg_time = total_time / num_iterations
print(
    f"Average time from original list of tuples {num_iterations} iterations: {avg_time:.6f} seconds"
)
```

results:
```
Average time from spatial_shapes tensor 1000 iterations: 0.000064 seconds
Average time from original list of tuples 1000 iterations: 0.000033 seconds
```

I think we should just drop the tensor entirely, given that we really are dealing with a 3x2 sized list, seems like tensors just add cognitive and performance overhead.

Reply:
I ended up dropping the spatial_shapes tensor that's passed around. I still have the original because `level_start_index` still exists, but I think we should just remove `level_start_index` in a separate PR.

Reply:
I address that change here: https://github.com/huggingface/transformers/pull/34498

Reply:
Ok, great! Thanks a lot for benchmarking it!

Reply:
no problem